Core/Tornado/Client/private/TornadoBaseClient.py

```diff
@@ -0,0 +1,569 @@
+          * if set, we remove the useCertificates (KW_USE_CERTIFICATES) in newKwargs
+
+        This method is just used to return information in case of error in the InnerRPCClient
+
+        WARNING: COPY/PASTE FROM Core/Diset/private/BaseClient
+    """
+    newKwargs = dict(self.kwargs)
+    # Remove useCertificates as the forwarder of the call will have to
+    # independently decide whether to use their cert or not anyway.
+    if 'useCertificates' in newKwargs:
+      del newKwargs['useCertificates']
+    return (self._destinationSrv, newKwargs)
+
+  def _request(self, retry=0, outputFile=None, **kwargs):
+    """
+      Sends the request to server
+
+      :param retry: internal parameters for recursive call. TODO: remove ?
+      :param outputFile: (default None) path to a file where to store the received data.
+                        If set, the server response will be streamed for optimization
+                        purposes, and the response data will not go through the
+                        JDecode process
+      :param **kwargs: Any argument there is used as a post parameter. They are detailed bellow.
+      :param method: (mandatory) name of the distant method
+      :param args: (mandatory) json serialized list of argument for the procedure
+
+
+
+      :returns: The received data. If outputFile is set, return always S_OK
+
+    """
+
+    # Adding some informations to send
+    if self.__extraCredentials:
+      kwargs[self.KW_EXTRA_CREDENTIALS] = encode(self.__extraCredentials)
+    kwargs["clientVO"] = self.vo
+
+    # Getting URL
+    url = self.__findServiceURL()
+    if not url['OK']:
+      return url
+    url = url['Value']
+
+    # Getting CA file (or skip verification)
+    verify = (not self.kwargs[self.KW_SKIP_CA_CHECK])
+    if verify and self.__ca_location:
+      verify = self.__ca_location
+
+    # getting certificate
+    if self.kwargs[self.KW_USE_CERTIFICATES]:
+      cert = Locations.getHostCertificateAndKeyLocation()
+    else:
+      cert = self.__proxy_location
+
+    # Do the request
+    try:
+      rawText = None
+
+      # Default case, just return the result
+      if not outputFile:
+        call = requests.post(url, data=kwargs,
+                             timeout=self.timeout, verify=verify,
+                             cert=cert)
+        # raising the exception for status here
+        # means essentialy that we are losing here the information of what is returned by the server
+        # as error message, since it is not passed to the exception
+        # However, we can store the text and return it raw as an error,
+        # since there is no guarantee that it is any JEncoded text
+        # Note that we would get an exception only if there is an exception on the server side which
+        # is not handled.
+        # Any standard S_ERROR will be transfered as an S_ERROR with a correct code.
+        rawText = call.text
+        call.raise_for_status()
+        return decode(rawText)[0]
```

Comment:
Is this safe in this order? Naively I would expect you to need `call.raise_for_status()` before trying to access the body:
```suggestion
        call.raise_for_status()
        return decode(call.text)[0]
```

Reply:
from the tests I could do and a quick glance at the code of `requests, yes. I do it here to be able to keep the message potentially returned by the server. If I `raise_for_status` before, I do not have access to `call` anymore.

Reply:
Of course I wasn't thinking properly. There is always a body, even if it's an error or empty.

You can still get the original request from the Exception that is raised (i.e. `e.response.text`).

Reply:
True ! I had not see that thanks